clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -63,6 +78,15 @@ export class CopybookDownloadService {
     ) {
       return true;
     }
+    try {
+      const res = await this.downloadCopybooksinProcessorGroups(
+        copybookName,
+        documentUri,
+      );
+      if (res) return res;
+    } catch (_error) {
+      return false;
+    }
```

Comment:
When did this start throwing and what?

Reply:
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2758/files#diff-faab25abb8052ae63940454cac14c6a4b0a4b1876118b4fa058ff8facdadffe8R149

throws err of getAllMembers failed requests

Reply:
Then is should be handled there.

Reply:
It is handled with returning that the download failed and the exception handled in `limitFailedRequests` with prompting to stop failed requests or continue. Do you think of any other exception needs to be handled ?

Reply:
We cannot just drop the error like nothing happened, it needs to be logged/shown to the user depending on the nature of the error.

Also as far as I can tell if the `downloadCopybooksinProcessorGroups` does not find the copybook in the provided library list, it fallback to using the settings.

Reply:
error logged , fixed fallback to user settings and resolved issue of download loop in the latest commits.